### PR TITLE
CAMEL-18127: implement adapter auto-configuration in the resume API

### PR DESCRIPTION
**Pull request by orpiske**
This is the first part of CAMEL-18127 and the work to make the resume
api simpler to use (see apache/camel-examples#92 for details). 

There are a few things missing in this PR, that will be completed in
subsequent PRs (otherwise it would be too big): 

- Update the remaining components to implement auto-configuration for
  them
- endpoint self-configuration (CAMEL-18128)
- improve the serialization (CAMEL-18148) 
- generate the adapter configuration automatically (CAMEL-18152)

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by orpiske** [comment]
> ❌ Finished component verification: **1 component(s) test failed** out of 7 component(s) tested

Test failure is unrelated to this change and a work-around is provided on #7690.

File: components/camel-atom/src/main/resources/org/apache/camel/resume/adapter [hunk 1]
```diff
@@ -0,0 +1 @@
+org.apache.camel.component.atom.UpdatedDateFilter
```
Comment (davsclaus) [inline components/camel-atom/src/main/resources/org/apache/camel/resume/adapter:1]: You need to have ASF license header on these files too

**Review by davsclaus — APPROVED** [review]

File: components/camel-atom/src/main/resources/org/apache/camel/resume/adapter (hunk 1 above)
Comment (orpiske) [reply components/camel-atom/src/main/resources/org/apache/camel/resume/adapter:1]: Thanks! I'll need to adjust the loader to filter out the license.

**Comment by orpiske** [comment]
Please don't merge it yet. It needs a fix to load the config files w/ license information.

**Pull request merged**